docs/selectors.md

```diff
@@ -84,9 +84,45 @@ Id engines are selecting based on the corresponding atrribute value. For example
 
 ### zs
 
-ZSelector is an experimental engine that tries to make selectors survive future refactorings. Example: `zs=div ~ "Login"`.
+Z-selector is an experimental engine to define selectors that can **survive future refactorings**. Changes in the UI markup can often cause existing UI selectors to break. While `data-*` attributes can be useful to establish contracts between app and test code, they can be an additional maintenance burden. It can also be difficult to introduce new attributes if you are testing third-party UI.
 
-TODO: write more.
+Z-selector aims to search elements with precision and resiliency against layout changes. It does not depend on strict nesting, and can localize element search through accessible selectors, like text content and type of element. It can then combine that with precise last-mile lookup.
+
+Z-selector selector uses the `~` combinator 
+(not to be confused with the [general sibling combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator)) to traverse the DOM hierarchy. Let's see how `zs="Foo" ~ "Bar" ~ "Baz"` works. From the root, it will look for a node with the text "Foo". It will then start climbing up the hierarchy, until it finds a node with text "Bar". From that node it will start climbing until it finds a node "Baz".
```

Comment:
CSS selectors are resolved from right to left. Is the same true for Z selector? If so, maybe we should explain it that way?

Reply:
Are they? :-)

Reply:
That’s what the #perfmatters pro-tip always said, and at least when I worked on the IE dev tools, Trident behaved that way. But you’re right, I should have posed my original comment as a question 😁

Reply:
correct me if i'm wrong: my understanding is the RTL is an implementation detail for internal parsing when searching for "style that applies to a particular node", and reading a CSS selector from LTR or RTL results in the same selection. this is true for the zs selector as well, and we can skip this level of detail in this doc.